### ISSUE TITLE
Issue with postdominator analysis

### ISSUE THREAD

**Issue by robdockins** [GaloisInc/llvm-pretty-bc-parser#1]
I think there is a bug in the postdominator analysis in that gives inaccurate results on switch statements.

Consider the attached C and LLVM assembly files: the only difference between them is that one uses a branch and the other uses a switch.  Control flow is otherwise identical.  I expect to see the same postdominator graph, but actually there are differences..

```
Right m <- parseBitCodeFromFile "asdf.bc"
[ (defName def,pdoms (buildCFG (defBody def))) | def <- modDefines m ]
```
Yields:
```
[(Symbol "switch_fun"
  ,[(BB#4,[])
  ,(BB#0,[BB#4])
  ,(BB#3,[BB#4])
  ,(BB#1,[BB#4,BB#3])
  ,(BB#2,[BB#4,BB#3])])
,(Symbol "branch_fun"
  ,[(BB#3,[])
  ,(BB#1,[BB#3])
  ,(BB#0,[BB#3])
  ,(BB#2,[BB#3])])
]
```

I'm not entirely sure how to interpret these, but I think the results for the switch are less precise and I expect them to be the same.

**Comment by robdockins** [comment]
asdf.c
```
#include <stdint.h>

uint8_t switch_fun(uint8_t in) {
 switch (in) {
 case 0:
   return 1;
 default:
   return 4;
 }
}

uint8_t branch_fun(uint8_t in) {
  if( in == 0 ) {
   return 1;
  } else {
   return 4;
  }
}
```

asdf.ll
```
; ModuleID = 'asdf.c'
target datalayout = "e-m:o-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-apple-macosx10.10.0"

; Function Attrs: nounwind ssp uwtable
define zeroext i8 @switch_fun(i8 zeroext %in) #0 {
  %1 = alloca i8, align 1
  %2 = alloca i8, align 1
  store i8 %in, i8* %2, align 1
  %3 = load i8* %2, align 1
  %4 = zext i8 %3 to i32
  switch i32 %4, label %6 [
    i32 0, label %5
  ]

; <label>:5                                       ; preds = %0
  store i8 1, i8* %1
  br label %7

; <label>:6                                       ; preds = %0
  store i8 4, i8* %1
  br label %7

; <label>:7                                       ; preds = %6, %5
  %8 = load i8* %1
  ret i8 %8
}

; Function Attrs: nounwind ssp uwtable
define zeroext i8 @branch_fun(i8 zeroext %in) #0 {
  %1 = alloca i8, align 1
  %2 = alloca i8, align 1
  store i8 %in, i8* %2, align 1
  %3 = load i8* %2, align 1
  %4 = zext i8 %3 to i32
  %5 = icmp eq i32 %4, 0
  br i1 %5, label %6, label %7

; <label>:6                                       ; preds = %0
  store i8 1, i8* %1
  br label %8

; <label>:7                                       ; preds = %0
  store i8 4, i8* %1
  br label %8

; <label>:8                                       ; preds = %7, %6
  %9 = load i8* %1
  ret i8 %9
}

attributes #0 = { nounwind ssp uwtable "less-precise-fpmad"="false" "no-frame-pointer-elim"="true" "no-frame-pointer-elim-non-leaf" "no-infs-fp-math"="false" "no-nans-fp-math"="false" "stack-protector-buffer-size"="8" "unsafe-fp-math"="false" "use-soft-float"="false" }

!llvm.ident = !{!0}

!0 = metadata !{metadata !"Apple LLVM version 6.1.0 (clang-602.0.53) (based on LLVM 3.6.0svn)"}
```

**Comment by elliottt** [comment]
The `brTargets` function from `llvm-pretty` didn't include a case for `Switch`, which I'm assuming is causing the difference.  I'm working on a bit of a refactoring to include that constructor, and simplify the implementation of `buildCFG`.

**Comment by elliottt** [comment]
This is fixed in ae1cbd1628ede8f9a174dc4a0176976a24beff25.
